py/desispec/preproc.py

```diff
@@ -597,7 +597,7 @@ def get_calibration_image(cfinder, keyword, entry, header=None):
             expid = header['EXPID']
             camera = header['CAMERA'].lower()
             if 'DESI_SPECTRO_REDUX' in os.environ and 'SPECPROD' in os.environ:
-                biasnight = findfile('biasnight', night, expid, camera)
+                biasnight = findfile('biasnight', night, expid, camera, readonly=True)
```

Comment:
Was this just because it was good to add readonly=True here, or was it actually necessary?

Reply:
Without this option, the code was crashing because I had started to run the pipeline with DESI_COMPRESSION=NONE and then resumed running with DESI_COMPRESSION=GZ. It's a case where compressed or uncompressed filenames work when the option readonly=True is set. This error would not have occurred if I had not changed the compression level in the middle of the processing of a night!